master/buildbot/status/web/status_json.py

```diff
@@ -1180,6 +1182,40 @@ def asDict(self, request):
         defer.returnValue(result)
 
 
+class BuildRequestJsonResource(JsonResource):
+    help = 'Gives information about build request'
+    pageTitle = 'BuildRequest Info'
+
+    def __init__(self, status):
+        JsonResource.__init__(self, status)
+
+    def getChild(self, path, request):
+        try:
+            build_request_id = int(path)
+        except (TypeError, ValueError):
+            return resource.NoResource('Invalid build request ID.')
+
+        if request.postpath and request.postpath[0] == 'build_number':
+            return BuildNumberForRequestJsonResource(self.status, build_request_id)
```

Comment:
Minor for readability:

We can follow by python zen and keep this function more flatten. I mean, we can move error path into `nested if` but success path just has `return` on end of the method. 

So in this case, we can write:
```
if not request.postpath or request.postpath[0] != 'build_number':
    return resource.NoResource('Resource not found.')

return BuildNumberForRequestJsonResource(self.status, build_request_id)
```

Reply:
@Mowinski Fixed :)